lib/Dialect/FIRRTL/Import/FIRAnnotations.cpp

```diff
@@ -1674,10 +1674,9 @@ bool circt::firrtl::scatterCustomAnnotations(
         if (!handleTarget(attr, i++, false))
           return false;
 
-      // Add the emitJSON attribute as we now have enough information to know if
-      // we are in the subcircuit or the main circuit.
-      if (emitJSON)
-        fields.append("emitJSON", UnitAttr::get(context));
+      // Add field indicating if we're the subcircuit or not.
+      fields.append("isSubCircuit", BoolAttr::get(context, isSubCircuit));
```

Comment:
Oh, it's mandatory but its value is important for indicating if we're in the main circuit or subcircuit.

We could make it a UnitAttr so that its absence indicates false (or rename and make it other way 'round), maybe that's what you meant, but for now I think it's good to explicitly specify this and to catch situations where we reach this point without adding it.  If that makes sense? :smile:.

Alternatively we could probably do the same sort of "is this the main or local circuit" check in this pass, to be honest.